07_safety/juvenile-arrests/juvenile-arrests-county-all.Rmd

```diff
@@ -0,0 +1,971 @@
+    race_of_arrestee = if_else(race_of_arrestee=="b", "black", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="i", "american indian/alaskan native", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="a", "asian", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="p", "native hawaiian or other pacific islander", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="u", "unknown", race_of_arrestee)
+  )
+
+## Recode ethnicity
+arrests <- arrests %>%
+  mutate(
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="h", "hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="n", "not of hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="u", "unknown", ethnicity_of_arrestee)
+  )
+
+```
+
+
+```{r}
+# The following code only pertains to the year 2015. I am leaving the code here (commented out) for the benefit of future updates that may pull historical data:
+
+# 2015 has 1953 observations with a race code of `f`. There is no codebook and I'm unclear what this means. Additionally, observations with a race code of `f` or `unknown` have ethnicity codes of `a`, `b`, `i`, and `w`. There are 3,877 observations with these ethnicity codes for the `unknown` race group.
+
+# # Check race and ethnicity codes for 2015
+# arrests %>%
+#   filter(year==2015) %>%
+#   group_by(race_of_arrestee, ethnicity_of_arrestee) %>%
+#   count()
+
+```
+
+58,052 observations have `unknown` ethnicity and 75,253 have missing ethnicity
+```{r}
+arrests %>% 
+  group_by(ethnicity_of_arrestee) %>% 
+  count()
+
+```
+
+
+### 3.4 Aggregate arrest counts to the reporting agency level
+Aggregate arrests by agency and finish race construction
+
+Note that this metric includes the following subgroups by race, sex, and age subgroups, so those subgroups are also aggregated in this section:
+    * Race: white, Black, Hispanic, Asian/other
+    * Sex: male, female
+    * Age: ages 10-14, ages 15-17
+```{r}
+arrests_agency <- arrests %>%
+  filter(age >= 10) %>%
+  group_by(year, ori) %>%
+  summarize(
+    arr_total = n(),
+    # arr_prop = sum(property==1),
+    # arr_viol = sum(violent==1),
+    # Race subgroups
+    race_aian = sum(race_of_arrestee == "american indian/alaskan native"),
+    race_asian = sum(race_of_arrestee == "asian"),
+    race_black = sum(race_of_arrestee == "black"),
+    race_nhpi = sum(race_of_arrestee == "native hawaiian or other pacific islander"),
+    race_unkno = sum(race_of_arrestee == "unknown"),
+    race_white = sum(race_of_arrestee == "white"),
+    ethn_hispanic = sum(ethnicity_of_arrestee == "hispanic origin"),
+    ethn_nonhisp = sum(ethnicity_of_arrestee == "not of hispanic origin"),
+    # Sex subgroups
+    sex_male = sum(sex_of_arrestee == "male"),
+    sex_female = sum(sex_of_arrestee == "female"),
```

Comment:
Thanks for flagging. The sex subgroup was not in my initial update of this metric, so I overlooked that the Group B observations have a value of `f` instead of `female`. I re-coded these observations in the same section where I re-code race and ethnicity, so it no longer needs fixed here.